File: COMPSCI-7007/AimToTen.java
public class AimToTen {

    public static int need(int[] marks) {
        double sum = 0;
        int ans = 0;
        for (int i = 0; i < marks.length; i++) {
            sum = sum + marks[i];
        }
        System.out.println("Sum " + marks.length);
        while (sum / (marks.length + ans) <= 9.5) {
            sum += 10;
            ans++;
        }
        if (ans - 1 == -1)
            return 0;
        return ans - 1;
    }

    public static void main(String[] args) {
        int[] marks = { 8, 9 };
        int minAssignments = need(marks);
        System.out.println("Assignments still required : " + minAssignments);
    }

}

File: CodeDump/AimToTen.js
var a = [10, 10, 10, 10, 10, 10, 10, 10, 10, 9];
var sum,
  ans = 0;

function calcAverage(a) {
  var result = 0;
  for (var i = 0; i < a.length; i++) {
    result += a[i];
  }
  return result / a.length;
}

function need(a) {
  var average = calcAverage(a); // O(n)
  if (average >= 9.5) {
    return ans;
  } else {
    a.push(10);
    ans++;
    need(a);
  }
}

var ans = need(a);
